File: app/src/main/java/soup/movie/data/soup/model/TimeTableRequest.java
package soup.movie.data.soup.model;

import java.util.HashMap;
import java.util.Map;

import soup.movie.data.core.QueryMapProvider;

public class TimeTableRequest implements QueryMapProvider {

    private String areaCode;
    private String theaterCode;
    private String movieCode;

    public TimeTableRequest(String areaCode, String theaterCode, String movieCode) {
        this.areaCode = areaCode;
        this.theaterCode = theaterCode;
        this.movieCode = movieCode;
    }

    public String getAreaCode() {
        return areaCode;
    }

    public String getTheaterCode() {
        return theaterCode;
    }

    public String getMovieCode() {
        return movieCode;
    }

    @Override
    public String toString() {
        return "TimeTableRequest{}";
    }

    @Override
    public Map<String, String> toQueryMap() {
        return new HashMap<>();
    }
}

File: data/src/main/java/soup/movie/data/model/TimeTableRequest.kt
package soup.movie.data.model

import java.util.HashMap

import soup.movie.data.base.QueryMapProvider

class TimeTableRequest(val theaterCode: String, val movieCode: String) : QueryMapProvider {

    override fun toString(): String {
        return "TimeTableRequest{" +
                "theaterCode='" + theaterCode + '\''.toString() +
                ", movieCode='" + movieCode + '\''.toString() +
                '}'.toString()
    }

    override fun toQueryMap(): Map<String, String> {
        return HashMap()
    }
}
